File: server_integration_tests/dart_client/test/fixtures.dart
import 'package:graphlink_server_integration_tests_dart_client/generated/graphlink.dart';

const _port = 9997;

String get httpUrl => 'http://localhost:$_port/graphql';
String get wsUrl => 'ws://localhost:$_port/graphql';

DefaultGraphLinkWebSocketAdapter newWsAdapter() =>
    DefaultGraphLinkWebSocketAdapter(url: wsUrl, reconnect: true, protocols: ['graphql-transport-ws']);

GraphLinkClient newClient() => GraphLinkClient(
      adapter: GraphLinkDioAdapter(url: httpUrl).call,
      wsAdapter: newWsAdapter(),
    );

File: server_integration_tests/kotlin/kotlin_client/client-app/src/test/kotlin/dev/graphlink/kotlinclienttest/Fixtures.kt
package dev.graphlink.kotlinclienttest

import dev.graphlink.kotlinclient.generated.client.DefaultGraphLinkClientAdapter
import dev.graphlink.kotlinclient.generated.client.DefaultGraphLinkWebSocketAdapter
import dev.graphlink.kotlinclient.generated.client.GraphLinkClient
import dev.graphlink.kotlinclient.generated.client.KotlinxSerializationGraphLinkJsonCodec

object Fixtures {

    val SERVER_PORT: Int = System.getProperty("serverPort")?.toIntOrNull() ?: 9994

    val HTTP_URL = "http://localhost:$SERVER_PORT/graphql"
    val WS_URL = "ws://localhost:$SERVER_PORT/graphql"

    fun newClient(): GraphLinkClient {
        val codec = KotlinxSerializationGraphLinkJsonCodec()
        return GraphLinkClient(
            adapter = DefaultGraphLinkClientAdapter(HTTP_URL),
            wsAdapter = DefaultGraphLinkWebSocketAdapter(WS_URL, protocols = listOf("graphql-transport-ws")),
            encoder = codec,
            decoder = codec,
        )
    }
}
